src/controller/python/chip/clusters/Attribute.py

```diff
@@ -246,6 +372,50 @@ def handleAttributeData(self, path: AttributePath, status: int, data: bytes):
             self._event_loop.call_soon_threadsafe(
                 self._handleAttributeData, path, status, data)
 
+    def _handleEventData(self, header: EventHeader, data: bytes):
+        try:
+            path = EventPath(
+                ClusterId=header.ClusterId, EventId=header.EventId)
+            eventType = _EventIndex.get(str(path), None)
+            eventValue = None
+            tlvData = chip.tlv.TLVReader(data).get().get("Any", {})
+            if eventType is None:
+                eventValue = ValueDecodeFailure(
+                    tlvData, LookupError("attribute schema not found"))
+            else:
+                try:
+                    eventValue = eventType(eventType.FromTLV(data))
+                except Exception as ex:
+                    logging.error(
+                        f"Error convering TLV to Cluster Object for path: Endpoint = {path.EndpointId}/Cluster = {path.ClusterId}/Event = {path.EventId}")
+                    logging.error(
+                        f"Failed Cluster Object: {str(eventType)}")
+                    logging.error(ex)
+                    eventValue = ValueDecodeFailure(
+                        tlvData, ex)
+
+                    # If we're in debug mode, raise the exception so that we can better debug what's happening.
+                    if (builtins.enableDebugMode):
+                        raise
+
+            with self._resLock:
+                if self._res['Events'].has_key(path):
```

Comment:
An event stream is exactly that - a stream of events. That means that even if there are duplicated events for the same path, each and every one of them should be visible.

This makes it behave like attributes that patch up the cache, which is not what we want.

Reply:
I am using the path as key, the value is a list which store a list of events with same path,
this logic here would firstly check if that event path has been stored, we would find that list and append event to that list.

Reply:
I don't think we should structure it that way - events should just be in a flat list sorted by the order in which they were emitted in time.

Reply:
done, thanks